docs/src/DeveloperGuide/WritingAdaptor.md

```diff
@@ -0,0 +1,264 @@
+# Tutorial: Writing a new adaptor
+
+In this tutorial we will develop a new QAT profile adaptor. We will make the
+adaptor a separate library which is dynamically loaded through the command line
+interface. All examples in this tutorial can be found in `AdaptorExamples`.
+
+Our first "adaptor" will be a "hello world" boilerplate which serves the purpose
+of giving the reader an understanding of how to define configurations for our
+adaptor. We will demonstrate how to use this adaptor from the command line.
+
+For our second adaptor, we will use a standard LLVM pass to demonstrate how to
```

Comment:
A link to the corresponding example may be helpful. 
I was confused by how the QAT native adaptor section fits into this outline here. It would be good to give a full index here with links to the corresponding sections.

Reply:
I've added an outline. Links will need to be added as an after thought PR since this PR is changing all the locations of what is located where (we need absolute URLs for links to work on the webpage when referring to source files).